rclpy/src/rclpy/node.cpp

```diff
@@ -62,4 +69,192 @@ get_node_namespace(py::capsule pynode)
+    py::list list_value = py::list(variant->double_array_value->size);
+    for (size_t i = 0; i < variant->double_array_value->size; ++i) {
+      list_value[i] = py::float_(variant->double_array_value->values[i]);
+    }
+    value = list_value;
+  } else if (variant->string_array_value) {
+    type_enum_value = rcl_interfaces__msg__ParameterType__PARAMETER_STRING_ARRAY;
+    py::list list_value = py::list(variant->string_array_value->size);
+    for (size_t i = 0; i < variant->string_array_value->size; ++i) {
+      list_value[i] = py::str(variant->string_array_value->data[i]);
+    }
+    value = list_value;
+  }
+
+  py::object type = pyparameter_type_cls(py::int_(type_enum_value));
+  return pyparameter_cls(pyname, type, value);
+}
+
+/// Populate a dict with parameters by node name
+/**
+ * \param[in] params the parameters for multiple nodes
+ * \param[in] pyparameter_cls the Parameter class
+ * \param[in] pyparameter_type_cls the Parameter.Type class
+ * \param[out] pynode_params a dictionary to populate with node names and parameters
+ */
+void
+_populate_node_parameters_from_rcl_params(
+  const rcl_params_t * params, py::object pyparameter_cls,
+  py::object pyparameter_type_cls, py::dict pynode_params)
+{
+  for (size_t i = 0; i < params->num_nodes; ++i) {
+    std::string node_name{params->node_names[i]};
+    if (node_name.empty()) {
+      throw std::runtime_error("expected node name to have at least one character");
+    }
+
+    // Make sure all node names start with '/'
+    if ('/' != node_name.front()) {
+      node_name.insert(node_name.begin(), '/');
+    }
+    auto pynode_name = py::str(node_name);
+
+    // make a dictionary for the parameters belonging to this specific node name
+    if (!pynode_params.contains(pynode_name)) {
+      pynode_params[pynode_name] = py::dict();
+    }
+    auto parameter_dict = pynode_params[pynode_name].cast<py::dict>();
+
+    rcl_node_params_t node_params = params->params[i];
+    for (size_t ii = 0; ii < node_params.num_params; ++ii) {
+      auto pyparam_name = py::str(node_params.parameter_names[ii]);
+
+      parameter_dict[pyparam_name] = _parameter_from_rcl_variant(
+        pyparam_name, &node_params.parameter_values[ii], pyparameter_cls, pyparameter_type_cls);
+    }
+  }
+}
+
+/// Populate a Python dict with node parameters parsed from CLI arguments
+/**
+ * \param[in] args CLI arguments to parse for parameters
+ * \param[in] pyparameter_cls the Parameter class
+ * \param[in] pyparameter_type_cls the Parameter.Type class
+ * \param[out] params_by_node_name A Python dict object to place parsed parameters into.
+ */
+void
+_parse_param_overrides(
+  const rcl_arguments_t * args, py::object pyparameter_cls,
+  py::object pyparameter_type_cls, py::dict pyparams_by_node_name)
+{
+  rcl_params_t * params = nullptr;
+  if (RCL_RET_OK != rcl_arguments_get_param_overrides(args, &params)) {
+    throw RCLError("failed to get parameter overrrides");
```

Comment:
@sloretz typo:  
```suggestion
    throw RCLError("failed to get parameter overrides");
```

Reply:
overrrides -> overrides in be8fa76bb89941318be2fe641d3b7b30cea1c37e